src/inference/dev_api/threading/ie_thread_safe_containers.hpp

```diff
@@ -53,28 +53,22 @@ template <typename T>
 class ThreadSafeBoundedPriorityQueue {
 public:
     ThreadSafeBoundedPriorityQueue() = default;
-    bool try_push(T value) {
-        _mutex.lock();
+    bool try_push(T&& value) {
         if (_capacity) {
-            _mutex.unlock();
             _pqueue.push(std::move(value));
-        } else {
-            _mutex.unlock();
         }
         return _capacity;
     }
     bool try_pop(T& value) {
         return _pqueue.try_pop(value);
     }
     void set_capacity(std::size_t newCapacity) {
```

Comment:
why 'new_capacity' is not 'bool'? It seems like there should be some check that number of pushed tasks doesn't exceed `new_capacity`, but I don't see such code.

Reply:
this class is a thin wrapper of PriorityQueue, which itself does not has limitation on the queue size, the capacity here is only used to control if able to continue to push to the queue from the caller logic.

Reply:
In this case there probably should be `_pqueue.set_capacity(newCapacity)`

Reply:
yes, but tbb priority queue does not have set_capacity interface, that's why I made this wrapper.